test/core/external/lib/mockZeroExOrderDataHandlerLibrary.spec.ts

```diff
@@ -0,0 +1,244 @@
+    let subjectOrderData: Bytes32;
+
+    beforeEach(async () => {
+      subjectOrderData = bufferArrayToHex(
+        bufferOrderHeader(
+          signatureLength,
+          zeroExOrderLength,
+          makerAssetDataLength,
+          takerAssetDataLength,
+        ).concat(bufferFillAmount(fillAmount)
+      ));
+    });
+
+    it("works", async () => {
+      const result = await zeroExExchangeWrapper.getFillAmount.callAsync(subjectOrderData);
+      expect(result).to.be.bignumber.equal(fillAmount);
+    });
+  });
+
+  describe("#getSignature", async () => {
+    let subjectOrderData: Bytes32;
+
+    beforeEach(async () => {
+      subjectOrderData = bufferArrayToHex(
+        bufferOrderHeader(
+          signatureLength,
+          zeroExOrderLength,
+          makerAssetDataLength,
+          takerAssetDataLength,
+        )
+        .concat(bufferFillAmount(fillAmount))
+        .concat(bufferSignature(signature)
+      ));
+    });
+
+    it("works", async () => {
+      const result = await zeroExExchangeWrapper.getSignature.callAsync(subjectOrderData);
+      expect(web3.toAscii(result)).to.equal(signature);
+    });
+  });
+
+  describe("#parseZeroExOrder", async () => {
+    let subjectOrderData: Bytes32;
+
+    beforeEach(async () => {
+      const zeroExOrderBuffer = bufferZeroExOrder(zeroExOrder);
+
+      subjectOrderData = bufferArrayToHex(zeroExOrderBuffer);
+    });
+
+    it("works", async () => {
+      const result = await zeroExExchangeWrapper.parseZeroExOrder.callAsync(
+        subjectOrderData,
+        makerAssetDataLength,
+        takerAssetDataLength
+      );
+
+      const [addresses, uints, makerAssetDataResult, takerAssetDataResult] = result;
+      const [makerResult, takerResult, feeRecipientResult, senderResult] = addresses;
+      const [
+        makerAssetAmountResult,
+        takerAssetAmountResult,
+        makerFeeResult,
+        takerFeeResult,
+        expirationResult,
+        saltResult,
+      ] = uints;
+
+      expect(ownerAccount).to.equal(makerResult);
+      expect(takerAddress).to.equal(takerResult);
+      expect(feeRecipientAddress).to.equal(feeRecipientResult);
+      expect(senderAddress).to.equal(senderResult);
+      expect(makerAssetAmount).to.be.bignumber.equal(makerAssetAmountResult);
+      expect(takerAssetAmount).to.be.bignumber.equal(takerAssetAmountResult);
+      expect(makerFee).to.be.bignumber.equal(makerFeeResult);
+      expect(takerFee).to.be.bignumber.equal(takerFeeResult);
+      expect(expirationTimeSeconds).to.be.bignumber.equal(expirationResult);
+      expect(salt).to.be.bignumber.equal(saltResult);
+      expect(makerAssetData).to.equal(web3.toAscii(makerAssetDataResult));
+      expect(takerAssetData).to.equal(web3.toAscii(takerAssetDataResult));
+    });
+  });
+
+  describe("#parseZeroExOrderData", async () => {
+    let subjectOrderData: Bytes32;
+
+    beforeEach(async () => {
+      const zeroExOrderBuffer = bufferZeroExOrder(zeroExOrder);
```

Comment:
move this setup into `test/utils/zeroExExchangeWrapper.ts`. give it a descriptive name